### ISSUE TITLE
'Method not found' for private methods when using enh-ruby-mode

### ISSUE THREAD

**Issue by RAbraham** [dgutov/robe#47]
Hi,
`robe-jump` works for for jumping to methods on other classes but when I do it on a private method, it shows all Modules which may have the method but not the private method?

**Comment by dgutov** [comment]
Works for me. Could you give an example?

Are you trying to jump to a private method in a different class?

**Comment by RAbraham** [comment]
Nope, private method in the same class. I am in the process of creating a sample rails project and testing it in a simple setting.

**Comment by dgutov** [comment]
When closing issues, you might want to leave a mention of what your actual problem was, or what made you think there was a problem.

**Comment by RAbraham** [comment]
Sorry, I am evaluating another IDE for a month now, so I won't have time to follow up on this but the problem still exists for me. I didn't want to keep the ticket open indefinitely. My bad.
Should I reopen it or keep it closed?

**Comment by dgutov** [comment]
Ah no, it's a good enough explanation. Please only reopen this when you have more details.

Good luck with that other editor.

**Comment by RAbraham** [comment]
Thank you for being supportive :). People like you and bbatsov make great packages for Emacs

**Comment by ledbettj** [comment]
Hi,

I'm seeing the same behavior.  Jumping to public methods works fine.  The following does not (rails 4.2.5, ruby 2.3.0):

```ruby
class TestModel < ActiveRecord::Base

  def baz
    foo # M-. here properly takes us to `foo`.
  end

  def foo
    bar # M-. says "Method not found"
  end

  private # If I remove this line and C-c C-k we can find `bar`.

  def bar # M-. here even says "Method not found"
    'hello world'
  end
end

```

robe version:

```
robe is an installed package.

     Status: Installed in ‘robe-20160427.552/’ (unsigned). Delete
    Version: 20160427.552
    Summary: Code navigation, documentation lookup and completion for Ruby
   Requires: inf-ruby-2.3.0

```

**Comment by dgutov** [comment]
@ledbettj Hi!

This works fine for me. I've just tried it in a new rails 4.2.5/ruby 2.3.0 project.

Have you tried pressing `C-c C-k` without removing `private`? Does navigation to `bar` start working then?

If not, please put the cursor inside the `foo` definition (on `bar` call) and evaluate (`M-:`) this expression: `(robe-context)`. What does it return?

Which version of Emacs are you using?

**Comment by ledbettj** [comment]
Thanks for the response!  `C-c C-k` without removing `private` has no effect.

`(robe-context)` on the bar call returns:

```
("TestModel" nil nil)
```

Emacs Version:
> GNU Emacs 25.0.93.1 (x86_64-apple-darwin15.4.0, NS appkit-1404.46 Version 10.11.4 (Build 15E65)) of 2016-04-25

Let me know if there's any other info that might be useful.

**Comment by dgutov** [comment]
Thanks. The result is wrong, it should be `("TestModel" t "foo")`.

Do you by any chance have an "alternative" `ruby-mode` package installed through ELPA? Please remove it.

**Comment by ledbettj** [comment]
That was it -- I use [enh-ruby-mode](https://github.com/zenspider/enhanced-ruby-mode) instead of the default `ruby-mode`.  Switching back to `ruby-mode` fixes.

**Comment by dgutov** [comment]
Hmm, that's a surprise. I didn't expect *it* to be the source of the problem.

I've filed an issue there.

**Comment by dgutov** [comment]
It's improved, and it'll probably fix the example by @ledbettj. But see my comment in the linked issue.

I've pushed a change to robe now that will allow it to use `enh-ruby-add-log-current-method`.

**Comment by dgutov** [comment]
Closing as third-party issue.

You (dear reader) might want to use `ruby-mode` or `ruby-ts-mode` instead.
